source/common/stats/thread_local_store.h

```diff
@@ -108,48 +108,37 @@ class ParentHistogramImpl : public MetricImpl<ParentHistogram> {
   const std::string bucketSummary() const override;
 
   // Stats::Metric
-  SymbolTable& symbolTable() override { return parent_.symbolTable(); }
+  SymbolTable& symbolTable() override;
   bool used() const override;
 
   // RefcountInterface
-  void incRefCount() override { refcount_helper_.incRefCount(); }
-  bool decRefCount() override { return refcount_helper_.decRefCount(); }
-  uint32_t use_count() const override { return refcount_helper_.use_count(); }
+  void incRefCount() override;
+  bool decRefCount() override;
+  uint32_t use_count() const override { return ref_count_; }
+
+  // Indicates that the ThreadLocalStore is shutting down, so no need to clear its histogram_set_.
+  void setShuttingDown(bool shutting_down) { shutting_down_ = shutting_down; }
+  bool shuttingDown() const { return shutting_down_; }
 
 private:
   bool usedLockHeld() const ABSL_EXCLUSIVE_LOCKS_REQUIRED(merge_lock_);
 
   Histogram::Unit unit_;
-  Store& parent_;
-  TlsScope& tls_scope_;
+  ThreadLocalStoreImpl& parent_;
```

Comment:
nit: s/parent_/thread_local_store_ or something as this is a little confusing already being a parent.